backend/flask_api.py

```diff
@@ -191,6 +192,62 @@ def delete_project(project_id):
         return make_response('Success')
 
 
+@app.route('/api/projects/<uuid:project_id>', methods=['PUT'])
+def update_project(project_id):
+    """Updates Project by ID
+
+    Args:
+        project_id, updated manifest in json/json5 format
+
+    Returns:
+        response: Success response
+                  or 404 if project is not found
+                  or 409 if project_id differs from manifestID
+                  or 500 in case of validation error
+    """
+    try:
+        res = coll.find_one({'_id': project_id})
+        if res is None:
+            raise ApiException("Project not found", 404)
+        elif request.is_json or request.headers['content-type'] == "application/json5":
+            if request.is_json:
+                manifest = request.json
+                if manifest['_id'] != str(project_id):
+                    raise ApiException("Updated project owns different id", 409)
+            else:
+                manifest = json5.loads(request.data.decode("utf-8"))
+                if '_id' in manifest:
+                    if manifest['_id'] != str(project_id):
+                        raise ApiException("Updated project owns different id", 409)
+            is_valid = validator.is_valid(manifest)
+            if is_valid:
+                print("manifest validated", file=sys.stderr)
+                manifest['_id'] = project_id
+                manifest['date_last_updated'] = time.strftime("%Y-%m-%d")
+                coll.find_one_and_replace({'_id': project_id}, manifest,
+                                          return_document=ReturnDocument.AFTER)
+                print("mongo replaced: ", file=sys.stderr)
+                es.index(index="projects-index", doc_type='Project',
+                         id=manifest["_id"], refresh=True, body={})
+                print("Successfully replaced content: ", file=sys.stderr)
+                print(manifest, file=sys.stderr)
+                return make_response('Success')
+            else:
+                v = validator.iter_errors(manifest)
+                if v is not None:
+                    validation_error = []
+                    for error in sorted(validator.iter_errors(manifest), key=str):
+                        print(error.message, file=sys.stderr)
```

Comment:
without this print it's possible to just
`validation_errors = [error for error in sorted(validator.iter_errors(manifest))]`